Lib/logging/handlers.py

```diff
@@ -187,15 +187,15 @@ def shouldRollover(self, record):
         Basically, see if the supplied record would cause the file to exceed
         the size limit we have.
         """
-        # See bpo-45401: Never rollover anything other than regular files
-        if os.path.exists(self.baseFilename) and not os.path.isfile(self.baseFilename):
-            return False
```

Comment:
Even with a local file-system this is a substantive chunk of the run time

<img width="1419" alt="Screenshot 2023-07-12 at 6 22 51 PM" src="https://github.com/python/cpython/assets/663432/1b873cd3-d0b0-4620-91ce-db9396b3a452">

Reply:
@bdraco I'm not familiar with the report generated by py-spy.  Are you seeing that the os.path.exists() and os.path.isfile() are expensive or the self.format()?  The self.format() is probably a different bug that needs fixed since rollover is formatting the message once to get its length and then emit later formats it again.

Reply:
I'd expect the format and emit to be expensive here. What is unexpected is the cost of `os.path.exists` and `os.path.isfile` which is solved by this PR.

Sorry for the lack of clarity.  This looks like a good fix and I wanted to point out the problem is not limited to NFS.

Reply:
Under normal usage, `self.baseFilename` doesn't change. Perhaps we just cache the result of the existence/is-file test when we open the file? Wouldn't that be a better approach?